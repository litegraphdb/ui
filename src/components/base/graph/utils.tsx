import { NodeData } from '@/lib/graph/types';
import { defaultNodeColor, nodeLightColorByType, universalNodeColors } from './constant';

const maxLegends = 16;

export const getLegendsForNodes = (nodes: NodeData[]) => {
  const priorityLegends = new Map<string, { legend: string; color: string }>();
  const customLegends = new Map<string, { legend: string; color: string }>();

  // Separate legends into priority (nodeLightColorByType) and custom
  nodes.forEach((node) => {
    if (Object.keys(nodeLightColorByType).includes(node.type)) {
      priorityLegends.set(node.type, { legend: node.type, color: nodeLightColorByType[node.type] });
    } else {
      // Only add if not already present to avoid changing colors for same legend
      if (!customLegends.has(node.type)) {
        customLegends.set(node.type, {
          legend: node.type,
          color: universalNodeColors[customLegends.size % universalNodeColors.length],
        });
      }
    }
  });

  const priorityLegendsSize = priorityLegends.size;
  // If there are nodes with unknown types, add "Unknown" to priority legends
  const isMoreThanMaxLegends = customLegends.size + priorityLegendsSize > maxLegends;

  // Convert to arrays for easier manipulation
  const priorityArray = Array.from(priorityLegends.values());
  const customArray = Array.from(customLegends.values()).slice(
    0,
    maxLegends - (priorityArray.length + (isMoreThanMaxLegends ? 1 : 0))
  );

  // Combine priority legends first, then custom legends
  const legends = [
    ...priorityArray,
    ...customArray,
    ...(isMoreThanMaxLegends ? [{ legend: 'Unknown', color: defaultNodeColor }] : []),
  ];
  const mapLegends: Record<string, { legend: string; color: string }> = {};
  legends.forEach((legend) => {
    mapLegends[legend.legend] = legend;
  });
  return mapLegends;
};
