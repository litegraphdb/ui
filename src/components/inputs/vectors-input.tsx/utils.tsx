export const convertVectorsToAPIRecord = (
  vectors?: Array<{ Model: string; Dimensionality: number; Content: string; Vectors: number[] }>
) => {
  return (
    vectors?.map((vector: any) => ({
      ...vector,
      Dimensionality: Number(vector.Dimensionality),
      Vectors: vector.Vectors || [],
    })) || []
  );
};
