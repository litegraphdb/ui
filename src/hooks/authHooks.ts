import { setAccessKey, setTenant, useGetUser } from '@/lib/sdk/litegraph.service';
import { useAppDispatch } from '@/lib/store/hooks';
import {
  logOut,
  storeToken,
  storeTenant,
  storeUser,
  storeAdminAccessKey,
} from '@/lib/store/litegraph/actions';
import { TenantMetaData, Token } from 'litegraphdb/dist/types/types';
import { setEndpoint, setAccessToken, sdk } from '@/lib/sdk/litegraph.service';
import { useGetTokenDetailsMutation } from '@/lib/store/slice/slice';
import toast from 'react-hot-toast';
import { globalToastId } from '@/constants/config';
import { localStorageKeys } from '@/constants/constant';

export const useFetchUserDetails = () => {
  const dispatch = useAppDispatch();
  const { fetchUser, isLoading, error } = useGetUser();

  const fetchUserDetails = async (userId: string, storeInRedux: boolean = true) => {
    const response = await fetchUser(userId);
    if (response?.GUID && storeInRedux) {
      dispatch(storeUser(response));
    }
  };
  return { fetchUserDetails, isLoading };
};

export const useCredentialsToLogin = () => {
  const dispatch = useAppDispatch();
  const loginWithCredentials = (token: Token, tenant: TenantMetaData) => {
    dispatch(storeToken(token));
    dispatch(storeTenant(tenant));
  };
  return loginWithCredentials;
};

export const useAdminCredentialsToLogin = () => {
  const dispatch = useAppDispatch();
  const loginWithAdminCredentials = (accessKey: string) => {
    dispatch(storeAdminAccessKey(accessKey));
  };
  return loginWithAdminCredentials;
};

export const useLogout = () => {
  const dispatch = useAppDispatch();

  const logOutFromSystem = (path?: string) => {
    dispatch(logOut(path));
  };
  return logOutFromSystem;
};

export const useApiKeyToLogin = () => {
  const loginWithCredentials = useCredentialsToLogin();
  const [getTokenDetails] = useGetTokenDetailsMutation();
  const loginWithApiKey = async (apikey: string, endpoint: string) => {
    setAccessToken(apikey);
    setEndpoint(endpoint);

    try {
      const tokenDetails = await getTokenDetails(apikey);
      if (tokenDetails?.data?.Tenant) {
        loginWithCredentials(tokenDetails.data, tokenDetails.data.Tenant);
        setTenant(tokenDetails.data.Tenant.GUID);

        // Store the API key separately for SDK configuration
        const tokenWithApiKey = {
          ...tokenDetails.data,
          Token: apikey, // Add the API key to the token object
        };

        localStorage.setItem(localStorageKeys.token, JSON.stringify(tokenWithApiKey));
        localStorage.setItem(localStorageKeys.tenant, JSON.stringify(tokenDetails.data.Tenant));
        localStorage.setItem(localStorageKeys.serverUrl, endpoint);

        // Also store user data if available
        if (tokenDetails.data.User) {
          localStorage.setItem(localStorageKeys.user, JSON.stringify(tokenDetails.data.User));
        }

        toast.success('Logged in successfully via SSO.', { id: globalToastId });
        return { success: true, tenant: tokenDetails.data.Tenant };
      }
      return { success: false, tenant: null };
    } catch (error) {
      console.error('Error: ', error);
      return { success: false, tenant: null };
    }
  };
  return { loginWithApiKey };
};
