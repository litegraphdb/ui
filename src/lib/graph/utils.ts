import type { NodeData, Point } from './types';

export function isPointInNode(point: Point, node: NodeData, radius: number = 10): boolean {
  const dx = point.x - node.x;
  const dy = point.y - node.y;
  return dx * dx + dy * dy <= radius * radius;
}

export function getNodeAtPosition(nodes: NodeData[], point: Point): NodeData | null {
  return nodes.find((node) => isPointInNode(point, node)) || null;
}

export function transformToOptions<T extends { GUID: string; name?: string; Name?: string }>(
  data?: T[] | null,
  labelField: keyof T = 'name' // Field to use for label in options
) {
  return data
    ? data?.map((item) => ({
        value: item.GUID,
        label: (item[labelField] as string) || item.Name || item.GUID,
      }))
    : [];
}
