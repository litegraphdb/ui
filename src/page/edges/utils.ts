import { EdgeType } from '@/lib/store/edge/types';
import { NodeType } from '@/lib/store/node/types';

// Utility to find the name by GUID
export const getNodeNameByGUID = (guid: string, nodesList: NodeType[]) => {
  if (!guid) return 'N/A';
  const node = nodesList.find((n: NodeType) => n.GUID === guid);
  return node ? node.Name : guid;
};

export const transformEdgeDataForTable = (edgesList: EdgeType[], nodesList: NodeType[]) => {
  return (
    edgesList?.map((data: EdgeType) => {
      return {
        ...data,
        FromName: getNodeNameByGUID(data.From, nodesList),
        ToName: getNodeNameByGUID(data.To, nodesList),
        key: data.GUID,
      } as EdgeType;
    }) || []
  );
};
