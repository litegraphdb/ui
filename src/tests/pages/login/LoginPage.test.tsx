import '@testing-library/jest-dom';
import React from 'react';
import { screen, fireEvent, waitFor } from '@testing-library/react';
import { createMockInitialState } from '../../store/mockStore';
import { renderWithRedux } from '@/tests/store/utils';
import { getServer } from '@/tests/server';
import { handlers } from './handler';
import { commonHandlers } from '@/tests/handler';
import userEvent from '@testing-library/user-event';
import { setEndpoint, useValidateConnectivity } from '@/lib/sdk/litegraph.service';
import { useGenerateTokenMutation, useGetTenantsForEmailMutation } from '@/lib/store/slice/slice';
import { useCredentialsToLogin } from '@/hooks/authHooks';
import toast from 'react-hot-toast';
import { mockTenantData, mockToken } from '../mockData';       
import LoginPage from '@/page/login/LoginPage';

const server = getServer([...commonHandlers, ...handlers]);

jest.mock('@/lib/sdk/litegraph.service', () => ({
  setEndpoint: jest.fn().mockResolvedValue(true),
  useValidateConnectivity: jest.fn().mockResolvedValue(true),
}));

// jest.mock('@/hooks/authHooks', () => ({
//   useCredentialsToLogin: jest.fn(),
// }));

// jest.mock('@/lib/store/slice/slice', () => ({
//   useGenerateTokenMutation: jest.fn(),
//   useGetTenantsForEmailMutation: jest.fn(),
// }));

// jest.mock('react-hot-toast', () => ({
//   error: jest.fn(),
//   success: jest.fn(),
// }));

// jest.mock('@/constants/constant', () => ({
//   localStorageKeys: {
//     token: 'token',
//     tenant: 'tenant',
//     serverUrl: 'serverUrl',
//   },
// }));

// // Mock UI components
// jest.mock('@/components/base/input/Input', () => {
//   return function MockLitegraphInput(props: any) {
//     return <input data-testid="litegraph-input" {...props} />;
//   };
// });

// jest.mock('@/components/base/select/Select', () => {
//   return function MockLitegraphSelect(props: any) {
//     return (
//       <select 
//         data-testid="litegraph-select" 
//         disabled={props.disabled}
//         onChange={(e) => props.onChange?.(e.target.value)}
//         {...props}
//       >
//         {props.options?.map((option: any) => (
//           <option key={option.value} value={option.value}>
//             {option.label}
//           </option>
//         ))}
//       </select>
//     );
//   };
// });

// jest.mock('@/components/base/button/Button', () => {
//   return function MockLitegraphButton(props: any) {
//     return (
//       <button 
//         data-testid="litegraph-button"
//         disabled={props.loading || props.disabled}
//         {...props}
//       >
//         {props.children}
//       </button>
//     );
//   };
// });

// jest.mock('@/components/base/flex/Flex', () => {
//   return function MockLitegraphFlex(props: any) {
//     return <div data-testid="litegraph-flex" {...props}>{props.children}</div>;
//   };
// });

// jest.mock('@/components/layout/LoginLayout', () => {
//   return function MockLoginLayout(props: any) {
//     return <div data-testid="login-layout">{props.children}</div>;
//   };
// });

// jest.mock('@/theme/theme', () => ({
//   LightGraphTheme: {
//     primary: '#1890ff',
//   },
// }));

// // Mock localStorage
// const mockLocalStorage = {
//   getItem: jest.fn(),
//   setItem: jest.fn(),
//   removeItem: jest.fn(),
//   clear: jest.fn(),
// };
// Object.defineProperty(window, 'localStorage', {
//   value: mockLocalStorage,
// });


describe('LoginPage', () => {
  beforeAll(() => {
    server.listen();
  });
  afterEach(() => server.resetHandlers());
  afterAll(() => server.close());

  it('should render the LoginPage', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<LoginPage />, initialState, undefined, true);

    await waitFor(() => {
      expect(screen.getByText('LiteGraph Server URL')).toBeVisible();
      expect(screen.getByPlaceholderText('https://your-litegraph-server.com')).toBeVisible();
    });
    expect(container).toMatchSnapshot('initial login step');

    const nextButton = screen.getByRole('button', { name: /next/i });
    await waitFor(() => {
      expect(nextButton).toBeVisible();
    });
  });

  it('should display server URL input field', async () => {
    const initialState = createMockInitialState();
    const { container } = renderWithRedux(<LoginPage />, initialState, undefined, true);

    const urlInput = screen.getByPlaceholderText('https://your-litegraph-server.com');
    await waitFor(() => {
      expect(urlInput).toBeVisible();
    });
    expect(urlInput).toMatchSnapshot();
  });

  describe('Initial Render and Step 0 (Server URL)', () => {
    it('should render login page with initial step (server URL)', () => {
      const initialState = createMockInitialState();
      renderWithRedux(<LoginPage />, initialState, undefined, true);

      expect(screen.getByLabelText('LiteGraph Server URL')).toBeInTheDocument();
      expect(screen.getByPlaceholderText('https://your-litegraph-server.com')).toBeInTheDocument();
      expect(screen.getByRole('button', { name: /next/i })).toBeInTheDocument();
    });

    it('should show step indicators with first step active', () => {
      const initialState = createMockInitialState();
      renderWithRedux(<LoginPage />, initialState);

      const stepIndicators = document.querySelectorAll('[class*="stepIndicator"]');
      expect(stepIndicators).toHaveLength(5);
    });

  //   it('should validate URL field and show error for invalid URL', async () => {
  //     const user = userEvent.setup();
  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'invalid-url');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByText('Please enter a valid URL!')).toBeInTheDocument();
  //     });
  //   });

  //   it('should reject non-HTTP/HTTPS URLs', async () => {
  //     const user = userEvent.setup();
  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'ftp://example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByText('Only HTTP or HTTPS URLs are allowed!')).toBeInTheDocument();
  //     });
  //   });

  //   it('should proceed to step 1 after successful server validation', async () => {
  //     const user = userEvent.setup();
  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(setEndpoint).toHaveBeenCalledWith('https://example.com');
  //       expect(useValidateConnectivity).toHaveBeenCalled();
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });
  //   });

  //   it('should not proceed if server validation fails', async () => {
  //     const user = userEvent.setup();
  //     const initialState = createMockInitialState();
  //         renderWithRedux(<LoginPage />, initialState);

  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(useValidateConnectivity).toHaveBeenCalled();
  //       // Should remain on step 0
  //       expect(screen.getByLabelText('LiteGraph Server URL')).toBeInTheDocument();
  //     });
  //   });
  // });

  // describe('Step 1 (Email Input)', () => {
  //   beforeEach(async () => {
  //     // Setup to reach step 1
  //   });

  //   const navigateToStep1 = async () => {
  //     const user = userEvent.setup();
  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });

  //     return user;
  //   };

  //   it('should render email input on step 1', async () => {
  //     await navigateToStep1();

  //     expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     expect(screen.getByPlaceholderText('Email')).toBeInTheDocument();
  //     expect(screen.getByRole('button', { name: /back/i })).toBeInTheDocument();
  //     expect(screen.getByRole('button', { name: /next/i })).toBeInTheDocument();
  //   });

  //   it('should validate email field', async () => {
  //     const user = await navigateToStep1();

  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'invalid-email');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByText('Please enter a valid email!')).toBeInTheDocument();
  //     });
  //   });

  //   it('should proceed to step 3 with single tenant', async () => {
  //     const user = await navigateToStep1();

  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'user@example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(useGetTenantsForEmailMutation).toHaveBeenCalledWith('user@example.com');
  //       expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
  //     });
  //   });

  //   it('should proceed to step 2 with multiple tenants', async () => {
  //     const user = await navigateToStep1();

  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'user@example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(useGetTenantsForEmailMutation).toHaveBeenCalledWith('user@example.com');
  //       expect(screen.getByLabelText('Tenants')).toBeInTheDocument();
  //     });
  //   });

  //   it('should handle tenant fetch error', async () => {
  //     const user = await navigateToStep1();

  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'user@example.com');

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //         expect(useGetTenantsForEmailMutation).toHaveBeenCalledWith('user@example.com');
  //       // Should remain on step 1
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });
  //   });
  // });

  // describe('Step 2 (Tenant Selection)', () => {
  //   const navigateToStep2 = async () => {
  //     const user = userEvent.setup();

  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     // Step 0: URL
  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });

  //     // Step 1: Email
  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'user@example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Tenants')).toBeInTheDocument();
  //     });

  //     return user;
  //   };

  //   it('should render tenant selection on step 2', async () => {
  //     await navigateToStep2();

  //     expect(screen.getByLabelText('Tenants')).toBeInTheDocument();
  //     expect(screen.getByTestId('litegraph-select')).toBeInTheDocument();
  //     expect(screen.getByRole('button', { name: /back/i })).toBeInTheDocument();
  //     expect(screen.getByRole('button', { name: /next/i })).toBeInTheDocument();
  //   });

  //   it('should display tenant options', async () => {
  //     await navigateToStep2();

  //     const selectElement = screen.getByTestId('litegraph-select');
  //     expect(selectElement).toBeInTheDocument();

  //     const options = screen.getAllByRole('option');
  //     expect(options).toHaveLength(mockTenantData.length);
  //     expect(screen.getByText('Tenant One')).toBeInTheDocument();
  //     expect(screen.getByText('Tenant Two')).toBeInTheDocument();
  //   });

  //   it('should proceed to step 3 after tenant selection', async () => {
  //     const user = await navigateToStep2();

  //     const selectElement = screen.getByTestId('litegraph-select');
  //     fireEvent.change(selectElement, { target: { value: 'tenant-1' } });

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
  //     });
  //   });

  //   it('should validate tenant selection is required', async () => {
  //     const user = await navigateToStep2();

  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByText('Please select a tenant!')).toBeInTheDocument();
  //     });
  //   });
  // });

  // describe('Step 3 (Password and Login)', () => {
  //   const navigateToStep3 = async () => {
  //     const user = userEvent.setup();

  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     // Navigate through steps
  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });

  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'user@example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
  //     });

  //     return user;
  //   };

  //   it('should render password input on step 3', async () => {
  //     await navigateToStep3();

  //     expect(screen.getByPlaceholderText('Password')).toBeInTheDocument();
  //     expect(screen.getByRole('button', { name: /back/i })).toBeInTheDocument();
  //     expect(screen.getByRole('button', { name: /login/i })).toBeInTheDocument();
  //   });

  //   it('should validate password field is required', async () => {
  //     const user = await navigateToStep3();

  //     const loginButton = screen.getByRole('button', { name: /login/i });
  //     await user.click(loginButton);

  //     await waitFor(() => {
  //       expect(screen.getByText('Please input your password!')).toBeInTheDocument();
  //     });
  //   });

  //   it('should handle successful login', async () => {
  //     const user = await navigateToStep3();

  //     const passwordInput = screen.getByPlaceholderText('Password');
  //     await user.type(passwordInput, 'password123');

  //     const loginButton = screen.getByRole('button', { name: /login/i });
  //     await user.click(loginButton);

  //     await waitFor(() => {
  //       expect(useGenerateTokenMutation).toHaveBeenCalledWith({
  //         email: 'user@example.com',
  //         password: 'password123',
  //         tenantId: 'tenant-1',
  //       });
  //       expect(localStorage.setItem).toHaveBeenCalledWith('token', JSON.stringify(mockToken));
  //       expect(localStorage.setItem).toHaveBeenCalledWith('tenant', JSON.stringify(mockTenantData[0]));
  //       expect(localStorage.setItem).toHaveBeenCalledWith('serverUrl', 'https://example.com');
  //       expect(useCredentialsToLogin).toHaveBeenCalledWith(mockToken, mockTenantData[0]);
  //     });
  //   });

  //   it('should handle login error when tenant not found', async () => {
  //     const user = await navigateToStep3();

  //     // Mock scenario where tenant is not found

  //     const passwordInput = screen.getByPlaceholderText('Password');
  //     await user.type(passwordInput, 'password123');

  //     const loginButton = screen.getByRole('button', { name: /login/i });
  //     await user.click(loginButton);

  //     await waitFor(() => {
  //       expect(toast.error).toHaveBeenCalledWith('Tenant not found');
  //     });
  //   });

  //   it('should handle token generation failure', async () => {
  //     const user = await navigateToStep3();

  //     const passwordInput = screen.getByPlaceholderText('Password');
  //     await user.type(passwordInput, 'password123');

  //     const loginButton = screen.getByRole('button', { name: /login/i });
  //     await user.click(loginButton);

  //     await waitFor(() => {
  //       expect(useGenerateTokenMutation).toHaveBeenCalled();
  //       // Should handle error gracefully
  //     });
  //   });
  // });

  // describe('Navigation', () => {
  //   it('should allow going back from step 1 to step 0', async () => {
  //     const user = userEvent.setup();     

  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage />, initialState);

  //     // Navigate to step 1
  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });

  //     // Go back to step 0
  //     const backButton = screen.getByRole('button', { name: /back/i });
  //     await user.click(backButton);

  //     expect(screen.getByLabelText('LiteGraph Server URL')).toBeInTheDocument();
  //     expect(screen.queryByRole('button', { name: /back/i })).not.toBeInTheDocument();
  //   });

  //   it('should preserve form data when navigating back and forth', async () => {
  //     const user = userEvent.setup();

  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage/>, initialState);

  //     // Fill step 0
  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });

  //     // Go back and verify URL is preserved
  //     await user.click(screen.getByRole('button', { name: /back/i }));
      
  //     const preservedUrlInput = screen.getByTestId('litegraph-input');
  //     expect(preservedUrlInput).toHaveValue('https://example.com');
  //   });
  // });

  // describe('Loading States', () => {
  //   it('should show loading state during server validation', async () => {
  //     const user = userEvent.setup();

  //     const initialState = createMockInitialState();
  //       renderWithRedux(<LoginPage />, initialState);

  //     const urlInput = screen.getByTestId('litegraph-input');
  //     expect(urlInput).toBeDisabled();

  //     const nextButton = screen.getByRole('button', { name: /loading/i });
  //     expect(nextButton).toBeDisabled();
  //   });

  //   it('should show loading state during tenant fetch', async () => {

  //     const initialState = createMockInitialState();
  //       renderWithRedux(<LoginPage />, initialState);

  //     const nextButton = screen.getByRole('button', { name: /loading/i });
  //     expect(nextButton).toBeDisabled();
  //   });

  //   it('should show loading state during token generation', async () => {

  //     const initialState = createMockInitialState();
  //       renderWithRedux(<LoginPage />, initialState);

  //     const loginButton = screen.getByRole('button', { name: /loading/i });
  //     expect(loginButton).toBeDisabled();
  //   });
  // });

  // describe('Form Validation Edge Cases', () => {
  //   it('should handle form validation errors gracefully', async () => {
  //     const user = userEvent.setup();
  //     const initialState = createMockInitialState();
  //         renderWithRedux(<LoginPage />, initialState);

  //     // Try to proceed without filling required field
  //     const nextButton = screen.getByRole('button', { name: /next/i });
  //     await user.click(nextButton);

  //     await waitFor(() => {
  //       expect(screen.getByText('Please enter the LiteGraph Server URL!')).toBeInTheDocument();
  //     });
  //   });

  //   it('should handle empty tenant response', async () => {
  //     const user = userEvent.setup();         

  //     const initialState = createMockInitialState();
  //     renderWithRedux(<LoginPage/>, initialState);

  //     // Navigate to step 1
  //     const urlInput = screen.getByTestId('litegraph-input');
  //     await user.type(urlInput, 'https://example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });

  //     const emailInput = screen.getByTestId('litegraph-input');
  //     await user.type(emailInput, 'user@example.com');
  //     await user.click(screen.getByRole('button', { name: /next/i }));

  //     // Should remain on step 1 since no tenants found
  //     await waitFor(() => {
  //       expect(screen.getByLabelText('Email')).toBeInTheDocument();
  //     });
  //   });
  });
});