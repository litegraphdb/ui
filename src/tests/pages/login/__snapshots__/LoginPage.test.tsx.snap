// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should display server URL input field 1`] = `
<input
  aria-required="true"
  class="ant-input ant-input-lg css-dev-only-do-not-override-98ntnt ant-input-outlined"
  data-testid="litegraph-input"
  id="url"
  placeholder="https://your-litegraph-server.com"
  type="text"
  value=""
/>
`;

exports[`LoginPage should render the LoginPage: initial login step 1`] = `
<div>
  <div
    class="userLoginPage ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-stretch ant-flex-vertical"
    style="gap: 20px;"
  >
    <div
      class="userLoginPageHeader mb pb pt pr pl ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center ant-flex-justify-space-between"
    >
      <img
        alt="Litegraph Logo"
        height="40"
        src="/favicon.png"
      />
      <div
        class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-center"
        style="gap: 10px;"
      >
        <svg
          color="black"
          fill="none"
          height="20"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="cursor: pointer; transform: rotate(90deg);"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <mask
            id="circle-mask-1"
          >
            <rect
              fill="white"
              height="100%"
              width="100%"
              x="0"
              y="0"
            />
            <circle
              fill="black"
              r="9"
            />
          </mask>
          <circle
            cx="12"
            cy="12"
            fill="black"
            mask="url(#circle-mask-1)"
          />
          <g
            stroke="currentColor"
            style="opacity: 1;"
          >
            <line
              x1="12"
              x2="12"
              y1="1"
              y2="3"
            />
            <line
              x1="12"
              x2="12"
              y1="21"
              y2="23"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="4.22"
              y2="5.64"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="18.36"
              y2="19.78"
            />
            <line
              x1="1"
              x2="3"
              y1="12"
              y2="12"
            />
            <line
              x1="21"
              x2="23"
              y1="12"
              y2="12"
            />
            <line
              x1="4.22"
              x2="5.64"
              y1="19.78"
              y2="18.36"
            />
            <line
              x1="18.36"
              x2="19.78"
              y1="5.64"
              y2="4.22"
            />
          </g>
        </svg>
        <a
          href="/login/admin"
        >
          <span
            class="ant-typography css-dev-only-do-not-override-98ntnt"
          >
            Login as Administrator
          </span>
        </a>
      </div>
    </div>
    <div
      class="loginTitle"
    >
      <h1
        class="ant-typography headingCommonStyles css-dev-only-do-not-override-98ntnt"
        style="font-weight: 600; font-size: 22px;"
      >
        Login
      </h1>
      <div
        class="ant-typography loginDescription css-dev-only-do-not-override-98ntnt"
      >
        Please enter your email and password to login
      </div>
    </div>
    <div
      class="loginBox"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-98ntnt ant-flex-align-stretch ant-flex-vertical"
        style="gap: 20px;"
      >
        <form
          class="ant-form ant-form-vertical css-dev-only-do-not-override-98ntnt"
        >
          <div
            class="ant-form-item css-dev-only-do-not-override-98ntnt"
          >
            <div
              class="ant-row ant-form-item-row css-dev-only-do-not-override-98ntnt"
            >
              <div
                class="ant-col ant-form-item-label css-dev-only-do-not-override-98ntnt"
              >
                <label
                  class="ant-form-item-required"
                  for="url"
                  title="LiteGraph Server URL"
                >
                  LiteGraph Server URL
                </label>
              </div>
              <div
                class="ant-col ant-form-item-control css-dev-only-do-not-override-98ntnt"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      aria-required="true"
                      class="ant-input ant-input-lg css-dev-only-do-not-override-98ntnt ant-input-outlined"
                      data-testid="litegraph-input"
                      id="url"
                      placeholder="https://your-litegraph-server.com"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="loginButtonContainer"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-98ntnt ant-btn-primary loginButton"
              type="submit"
            >
              <span>
                Next
              </span>
            </button>
          </div>
        </form>
        <div
          class="stepIndicatorContainer"
        >
          <div
            class="stepIndicator"
            style="background-color: rgb(85, 143, 66);"
          />
          <div
            class="stepIndicator"
            style="background-color: rgb(217, 217, 217);"
          />
          <div
            class="stepIndicator"
            style="background-color: rgb(217, 217, 217);"
          />
          <div
            class="stepIndicator"
            style="background-color: rgb(217, 217, 217);"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
